GeeksCoreLibrary/Modules/HealthChecks/Services/DatabaseHealthService.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using GeeksCoreLibrary.Core.Models;
+using GeeksCoreLibrary.Modules.Databases.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+
+namespace GeeksCoreLibrary.Modules.HealthChecks.Services;
+public class DatabaseHealthService : IHealthCheck
+{
+    private readonly IDatabaseConnection databaseConnection;
+    private readonly HttpContext httpContext;
+    private readonly HealthChecksSettings healthChecksSettings;
+    
+    public DatabaseHealthService(IDatabaseConnection databaseConnection, IHttpContextAccessor httpContextAccessor, IOptions<HealthChecksSettings> healthChecksSettings)
+    {
+        this.databaseConnection = databaseConnection;
+        httpContext = httpContextAccessor.HttpContext;
+        this.healthChecksSettings = healthChecksSettings.Value;
+    }
+    
+    /// <inheritdoc />
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context,
+        CancellationToken cancellationToken = new CancellationToken())
+    {
+        databaseConnection.ClearParameters();
+       
+        // A new query to check the max active connections from the DatabaseHealthCheck
+        var query =
+            $"SELECT COUNT(*) AS active_connections FROM information_schema.PROCESSLIST;";
+        var datatable = await databaseConnection.GetAsync(query);
+
+        if (datatable.Rows.Count == 0)
+        {
+            return await Task.FromResult(HealthCheckResult.Unhealthy("No data found"));
+        }
```

Comment:
This if statement can be removed, because a `COUNT` query will **always** return 1 result. So the code will never reach the inside of this statement. (If the table is empty, the query will return the result "0".)